### PR TITLE
* built-in: add "Sin(x)" function

### PR DESCRIPTION
**Pull request by alston111111**
* built-in: add "Sin(x)" function
Take a first try...

### CODE REVIEW

**Comment by coocood** [comment]
@alston111111 
Please fix CI.

**Comment by alston111111** [comment]
hi, I've met a problem:
I think I've passed the ./expression/builtin_math_test.go test, but stucked at "./plan/typeinferer_test.go" .
I've tried, but yet cannot figure out what was wrong. Could you please give a hint?
HERE is some detail: (for adding Sin function)
*** in ./plan/typeinferer_test.go, I added (the middle line, below):
		{"PI() + 0.000000000000000000", mysql.TypeDouble, charset.CharsetBin},
+		{"SIN(0)", mysql.TypeDouble, charset.CharsetBin},  // this was added for 'sin' type inferer
		{"ACOS(1)", mysql.TypeDouble, charset.CharsetBin},
*** in ./plan/typeinferer.go, I added:
+	case "sin":     // I've checked that ast.Sin == "sin"
+ 		tp = types.NewFieldType(mysql.TypeDouble)

The build log:
===================================
./expression/builtin_math_test.go
./plan/typeinferer_test.go
make: *** [check] Error 1
===================================

**Comment by alston111111** [comment]
Finally, I've found what was wrong with the CI on check stage and managed to fix it !!!

Just record it here _(I was almost new to GO and TiDB, so it spent me a lot of time to find the reason_. So excited!).
:::: The reason was **trying to commit the updates on the web directly and gofmt might return an error in the check stage.**
Not sure if any one has met this problem.

==> 1. If I tried to commit the updates on the web page (without cloning the project to local machine), then the CI check used gofmt to check indentations in the source files or sth. and modified files that are required to format.
==> 2. This resulted in the error "make: *** [check] Error 1".

==> 3. After I had cloned the project to my local machine and modified some files (sure the builds were OK), the command "make dev" might have a chance to get an "check error".
==> 4. The later "make dev" would return successfully all the way until the next time I made another change.
This result was similar to which I got After commiting updates on the web.

==> So a proper way is: (I've checked the Makefile, the required env vars and paths in detail)
1. Fork a copy in my own repository.
2. Clone my repository to the local repository.
3. Do some work and Merge locally.
**4. Execute "make dev" at least ## twice until you get no check error on gofmt.**
5. Commit and push and wait ...
6. If error ocurrs, repeat 4 and 5.

"""
./expression/builtin_math_test.go
./plan/typeinferer_test.go
make: *** [check] Error 1
"""

File: Makefile [hunk 1]
```diff
@@ -19,8 +19,8 @@ MAC       := "Darwin"
 PACKAGES  := $$(go list ./...| grep -vE 'vendor')
 FILES     := $$(find . -name '*.go' | grep -vE 'vendor')
 
-LDFLAGS += -X "github.com/pingcap/tidb/util/printer.TiDBBuildTS=$(shell date -u '+%Y-%m-%d %I:%M:%S')"
-LDFLAGS += -X "github.com/pingcap/tidb/util/printer.TiDBGitHash=$(shell git rev-parse HEAD)"
+LDFLAGS += -X "github.com/alston111111/tidb/util/printer.TiDBBuildTS=$(shell date -u '+%Y-%m-%d %I:%M:%S')"
```
Comment (coocood) [inline Makefile]: Please do not modify the Makefile.

You can put your fork in `github.com/pingcap` directory, then `make dev`.

Comment (alston111111) [inline Makefile]: Oh. Sorry about that.  Makefile has been changed back.

I was supposed to change it back before the push, but I forgot to.
I had already put my work in ${some_path}/github.com/pingcap.
And I will not do this again and I will pay attention to it.

Thank you.

**Comment by coocood** [comment]
LGTM
Thank you for your contribution!

**Comment by shenli** [comment]
LGTM
Thanks! @alston111111

**Pull request merged**